src/reducers/deleteThought.ts

```diff
@@ -35,8 +37,18 @@ const deleteThought = (state: State, { path }: { path?: Path }) => {
   if (!path) return state
 
   // same as in newThought
-  const contextChain = splitChain(state, path)
   const showContexts = isContextViewActive(state, pathToContext(contextOf(path)))
+  if (showContexts) {
+    // Get thought in ContextView
+    const thoughtInContextView = head(contextOf(path))
+    // Get context from which we are going to delete thought
+    const context = getContexts(state, thoughtInContextView.value).map(({ context }) => context).find(context => head(context) === headValue(path!))
```

Comment:
Similar: It is inefficient to `map` across the entire array when `find` naturally short circuits. I recommend combining into `find`.